span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -456,21 +454,21 @@ private static void setResponseStatusCode(
         .ifPresent(statusCode -> httpBuilder.getResponseBuilder().setStatusCode(statusCode));
   }
 
-  private static boolean isValidUrl(String s) {
+  private static boolean isValidUri(String s) {
     try {
-      new URL(s);
-    } catch (MalformedURLException e) {
+      new URI(s);
```

Comment:
We can continue using a check using URL with allowing both absolute and relative - via - `isAbsoluteURL`